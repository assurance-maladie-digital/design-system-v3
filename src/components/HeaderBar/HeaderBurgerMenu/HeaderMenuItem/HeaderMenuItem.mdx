import { Controls, Canvas, Meta, Source } from "@storybook/blocks";
import * as HeaderMenuItemStories from "./HeaderMenuItem.stories";

<Meta of={HeaderMenuItemStories} />

# HeaderMenuItem

Le composant `HeaderMenuItem` est un composant de menu qui permet de wrapper un lien de navigation. Cela peut être un lien `a`, `router-link` ou `nuxt-link`.
Ce composant doit être utilisé dans un composant `HeaderMenuSection` afin de respecter la structure du menu et de garantir la bonne accessibilité de celui ci.

<Source
  dark
  code={`
<HeaderMenuSection>
	<HeaderMenuItem>
		<a href="/">Home</a>
	</HeaderMenuItem>
	<HeaderMenuItem>
		<a href="/about">About</a>
	</HeaderMenuItem>
	<HeaderMenuItem>
		<RouterLink to="/services">Services</RouterLink>
	</HeaderMenuItem>
	<HeaderMenuItem>
		<NuxtLink to="/contact">Contact</NuxtLink>
	</HeaderMenuItem>
</HeaderMenuSection>
`}
/>

Pour plus de détails sur le contexte d'utilisation de ce composant, veuillez consulter le composant `HeaderBurgerMenu`.


<Canvas of={HeaderMenuItemStories.Default} />

## API

<Controls of={HeaderMenuItemStories.Default} />
