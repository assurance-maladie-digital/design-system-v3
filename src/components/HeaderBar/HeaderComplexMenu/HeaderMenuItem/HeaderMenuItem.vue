<script setup lang="ts">

</script>

<template>
	<li class="header-menu-item">
		<slot />
	</li>
</template>

<style lang="scss" scoped>
	@use "@/assets/tokens.scss" as *;

	.header-menu-item {
		color: $primary-base;
		list-style-type: none;
		margin: 0;
		padding: 0;
		min-height: 44px; // accessibility requirement
		font-weight: 700;

		> :deep(a) {
			display: flex;
			flex-direction: column;
			padding: 16px 50px 16px 20px;
			text-decoration: none;
			color: currentColor;

			&:hover {
				text-decoration: underline;
			}

			&:visited {
				color: currentColor;
			}

			&::first-letter {
				text-transform: uppercase;
			}
		}
	}

	.header-menu-item:hover {
		background-color: $primary-base;
		color: $neutral-white;

		> :deep(a > *) {
			color: $neutral-white !important;
		}
	}
</style>
