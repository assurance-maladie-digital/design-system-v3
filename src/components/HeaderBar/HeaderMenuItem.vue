<script setup lang="ts">

</script>

<template>
	<li class="header-menu-item">
		<slot />
	</li>
</template>

<style lang="scss" scoped>
	@use "@/assets/tokens.scss" as *;

	.header-menu-item {
		color: $primary-base;
		list-style-type: none;
		margin: 0;
		padding: 0;
		min-height: 44px; // accessibility requirement
		font-weight: 700;
		text-transform: capitalize;

		> :deep(a) {
			display: block;
			padding: 16px 50px 16px 20px;
			text-decoration: none;

			&:hover {
				text-decoration: underline;
			}

			&:focus, &:active, &:visited {
				color: currentColor;
			}
		}
	}

	.header-menu-item:hover {
		background-color: $primary-base;
		color: $neutral-white;
	}
</style>
