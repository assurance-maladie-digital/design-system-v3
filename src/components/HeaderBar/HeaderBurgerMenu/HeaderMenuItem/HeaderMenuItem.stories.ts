import type { Meta, StoryObj } from '@storybook/vue3'
import HeaderMenuItem from './HeaderMenuItem.vue'
import HeaderMenu from '../HeaderBurgerMenu.vue'
import HeaderBar from '../../HeaderBar.vue'

const meta = {
	component: HeaderMenuItem,
	parameters: {
		layout: 'fullscreen',
	},
} satisfies Meta<typeof HeaderMenuItem>

export default meta

type Story = StoryObj<typeof meta>

export const Default: Story = {
	args: {
		default: 'lorem ipsum',
	},
	render: (args) => {
		return {
			components: { HeaderMenuItem, HeaderMenu, HeaderBar },
			setup() {
				return { args }
			},
			template: `
				<HeaderBar>
					<template #menu>
						<HeaderMenu>
							<HeaderMenuSection>
								<HeaderMenuItem>
									<a>{{ args.default }}</a>
								</HeaderMenuItem>
							</HeaderMenuSection>
						</HeaderMenu>
					</template>
				</HeaderBar>
			`,
		}
	},
	play: async ({ canvasElement }) => {
		const menuBtn = canvasElement.querySelector('button')
		setTimeout(() => {
			menuBtn!.click()
		}, 1000)
	},
}
