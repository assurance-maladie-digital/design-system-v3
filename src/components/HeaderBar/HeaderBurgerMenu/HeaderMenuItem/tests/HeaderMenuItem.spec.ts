import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import HeaderMenuItem from '../HeaderMenuItem.vue'

describe('HeaderMenuItem', () => {
	it('should render the component', async () => {
		const wrapper = mount(HeaderMenuItem, {
			slots: {
				default: '<a>Test</a>',
			},
		})

		expect(wrapper.html()).toMatchSnapshot()
		expect(wrapper.find('a').text()).toBe('Test')
	})
})
