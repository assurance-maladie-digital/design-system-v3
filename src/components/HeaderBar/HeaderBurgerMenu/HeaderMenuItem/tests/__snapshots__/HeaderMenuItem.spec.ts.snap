// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeaderMenuItem > should render the component 1`] = `"<li data-v-bc776fc3="" class="header-menu-item"><a>Test</a></li>"`;
